platform/security/certificate/security-crl-generator/src/main/java/org/codice/ddf/security/crl/generator/CrlGenerator.java

```diff
@@ -0,0 +1,216 @@
+      new AbsolutePathResolver("etc/ws-security/server/encryption.properties").getPath();
+
+  @VisibleForTesting
+  static String serverSignaturePropertiesLocation =
+      new AbsolutePathResolver("etc/ws-security/server/signature.properties").getPath();
+
+  static String crlFileLocation = new AbsolutePathResolver("etc/keystores").getPath();
+
+  static final String DEM_CRL = "/localCrl.crl";
+
+  static final String PEM_CRL = "/localCrl.pem";
+
+  private static final PropertiesLoader PROPERTIES_LOADER = PropertiesLoader.getInstance();
+
+  private static final String HTTPS = "https://";
+
+  private static final int INITIAL_DELAY = 0;
+
+  private static final int SCHEDULER_INTERVAL = 30;
+
+  private final ClientFactoryFactory factory;
+
+  private final ScheduledExecutorService scheduler;
+
+  private String crlLocationUrl;
+
+  private boolean crlByUrlEnabled;
+
+  private ScheduledFuture<?> handle;
+
+  public CrlGenerator(ClientFactoryFactory factory) {
+    this.factory = factory;
+    this.scheduler =
+        Executors.newSingleThreadScheduledExecutor(
+            StandardThreadFactoryBuilder.newThreadFactory("crlThread"));
+  }
+
+  /** Pulls down the CRL file provided via URL and writes it locally. */
+  @Override
+  public synchronized void run() {
+
+    if (crlByUrlEnabled && crlLocationUrl != null && crlLocationUrl.startsWith(HTTPS)) {
+
+      try {
+        SecureCxfClientFactory cxfClientFactory =
+            factory.getSecureCxfClientFactory(crlLocationUrl, WebClient.class);
+        WebClient client = cxfClientFactory.getWebClient();
+        Response response = client.get();
+
+        Object entity = response.getEntity();
+        if (entity instanceof InputStream) {
+          try (ByteArrayInputStream inputStream =
+              IOUtils.loadIntoBAIS((InputStream) response.getEntity())) {
+
+            byte[] buffer = new byte[inputStream.available()];
+            int read = inputStream.read(buffer);
+
+            if (read > 0) {
+              File crlFile;
+              String localCrlPath;
+
+              if (Strings.fromByteArray(buffer).contains("-----BEGIN")) {
+                localCrlPath = crlFileLocation + PEM_CRL;
+                crlFile = new File(crlFileLocation + PEM_CRL);
+              } else {
+                localCrlPath = crlFileLocation + DEM_CRL;
+                crlFile = new File(localCrlPath);
+              }
+
+              try (OutputStream outStream = new FileOutputStream(crlFile)) {
+                outStream.write(buffer);
+                SecurityLogger.audit(
+                    "Copied the content of the CRl at {} to the local CRL at {}.",
+                    crlLocationUrl,
+                    localCrlPath);
+
+                setCrlFileLocationInPropertiesFile(localCrlPath);
+              }
+            }
+          }
+        }
+      } catch (IOException e) {
+        LOGGER.warn("Unable to copy CRL to local CRL.");
```

Comment:
❓ Should we include a little more diagnostic information, since we're swallowing the exception? Maybe include `e.getMessage()`.